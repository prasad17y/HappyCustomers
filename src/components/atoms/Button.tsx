import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  TouchableOpacityProps,
  StyleProp,
  ViewStyle,
} from 'react-native';

interface ButtonProps extends TouchableOpacityProps {
  style?: StyleProp<ViewStyle>;
}

const Button: React.FC<ButtonProps> = ({children, style, ...props}) => {
  return (
    <TouchableOpacity style={[styles.button, style]} {...props}>
      {children}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Button;
