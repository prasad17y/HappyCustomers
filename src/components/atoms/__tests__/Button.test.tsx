import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react-native';
import Button from '../Button';
import {Text} from 'react-native';

describe('Button', () => {
  it('should render children correctly', () => {
    render(
      <Button>
        <Text>Click Me</Text>
      </Button>,
    );
    expect(screen.getByText('Click Me')).toBeTruthy();
  });

  it('should handle onPress events', () => {
    const onPressMock = jest.fn();
    render(
      <Button onPress={onPressMock}>
        <Text>Click Me</Text>
      </Button>,
    );

    fireEvent.press(screen.getByText('Click Me'));
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('should match the snapshot', () => {
    const tree = render(
      <Button>
        <Text>Snapshot Button</Text>
      </Button>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
