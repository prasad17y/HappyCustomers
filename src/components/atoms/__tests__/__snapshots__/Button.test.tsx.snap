// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text>
    Snapshot Button
  </Text>
</View>
`;
